crates/core/tedge_agent/src/tedge_operation_converter/actor.rs

```diff
@@ -82,43 +78,54 @@ impl TedgeOperationConverterActor {
     }
 
     async fn publish_operation_capabilities(&mut self) -> Result<(), RuntimeError> {
-        let restart_capability =
-            RestartCommand::capability_message(&self.mqtt_schema, &self.device_topic_id);
-        Ok(self.mqtt_publisher.send(restart_capability).await?)
+        let capabilities = [
+            RestartCommand::capability_message(&self.mqtt_schema, &self.device_topic_id),
+            SoftwareListCommand::capability_message(&self.mqtt_schema, &self.device_topic_id),
+            SoftwareUpdateCommand::capability_message(&self.mqtt_schema, &self.device_topic_id),
+        ];
+        for capability in capabilities {
+            self.mqtt_publisher.send(capability).await?
+        }
+        Ok(())
     }
 
     async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), RuntimeError> {
-        match message.topic.name.as_str() {
-            "tedge/commands/req/software/list" => {
-                match SoftwareListRequest::from_slice(message.payload_bytes()) {
-                    Ok(request) => {
-                        self.software_sender
-                            .send(SoftwareRequest::SoftwareListRequest(request))
-                            .await?;
-                    }
-                    Err(err) => error!("Incorrect software list request payload: {err}"),
+        match self.mqtt_schema.entity_channel_of(&message.topic) {
+            // Software List
+            Ok((
+                target,
+                Channel::Command {
+                    operation: OperationType::SoftwareList,
+                    cmd_id,
+                },
+            )) => match SoftwareListCommand::try_from(target, cmd_id, message.payload_bytes()) {
+                Ok(Some(cmd)) => {
+                    self.software_sender.send(cmd.into()).await?;
                 }
-            }
-            "tedge/commands/req/software/update" => {
-                match SoftwareUpdateRequest::from_slice(message.payload_bytes()) {
-                    Ok(request) => {
-                        self.software_sender
-                            .send(SoftwareRequest::SoftwareUpdateRequest(request))
-                            .await?;
-                    }
-                    Err(err) => error!("Incorrect software update request payload: {err}"),
+                Ok(None) => {
+                    // The command has been fully processed
                 }
-            }
-            _ => {
-                // Not a tedge/commands !
-            }
-        }
+                Err(err) => error!("Incorrect software-list request payload: {err}"),
```

Comment:
To keep the consistency, use `_` instead of `-`. We use underscore for the predefined command types: https://thin-edge.github.io/thin-edge.io/next/references/mqtt-api/#command-examples

```suggestion
                Err(err) => error!("Incorrect software_list request payload: {err}"),
```

Reply:
Fixed by https://github.com/thin-edge/thin-edge.io/pull/2307/commits/3e8d8f8615f5cf4e012601bd23d68634f6e7384a